src/crypto/util.ts

```diff
@@ -27,8 +27,13 @@ export function legacyDeriveChildKey(
         password.byteOffset + seed.length,
         8
     );
-    view.setInt32(0, index);
-    view.setInt32(4, index >= 0 ? 0 : 0xFF);
+    if (index === 0xFFFFFFFFFF) {
+        view.setInt32(0, 0xFF);
+        view.setInt32(4, index >>> 32);
```

Comment:
I don't think this does what you think it does. `index >>> 32` would be `0xFF`. If `index` is `0xFFFFFFFFFF` then why not just set the value directly `view.setInt32(4, 0xFFFFFFFF)`

Reply:
`index >>> 32` would be `0xFFFFFFFF` here.

1. Shifts operate on 32-bit numbers
2. Only 5 bits are effective, so `>>> 32` is equivalent to  `>>> 0`.

This way of writing down `0xFFFFFFFF` is confusing, as that's not what one might expect for `0xFFFFFFFFFF >>> 32` to be.

Refs: https://262.ecma-international.org/11.0/#sec-numeric-types-number-unsignedRightShift
Refs: https://262.ecma-international.org/11.0/#table-numeric-type-ops


@danielakhterov @andrix10 this likely should be fixed somehow, I just don't know what's the correct value there.

Reply:
Which is the same as `view.setInt32(4, -1);` btw.

Reply:
@ChALkeR I forget when, but we did eventually stumble onto this issue when writing the v2 SDK (this PR is for v1). The current implementation in v2 is
```javascript
    if (index === 0xffffffffff) {
        view.setInt32(seed.length + 0, 0xff);
        view.setInt32(seed.length + 4, index >>> 32);
    } else {
        view.setInt32(seed.length + 0, index < 0 ? -1 : 0);
        view.setInt32(seed.length + 4, index);
    }
```
which tries to maintain backwards compatibility. Also, this is _legacy_ derive. Users are encouraged not to use these methods unless they created keys using the old v1 SDK or created keys using Hedera's legacy mobile application.

Reply:
@danielakhterov Yes, this is the same code chunk.
For `index === 0xffffffffff`, `index >>> 32` is `0xffffffff`, which is `-1`.
Not `0xff`, as one might think while reading this code.

Reply:
So you're suggesting we use `-1` instead of `index >>> 32` since it's more clear with `-1`? That sounds reasonable to me.

Reply:
Yes.

Reply:
Can you make a PR for this?

Reply:
I will. https://github.com/hashgraph/hedera-sdk-js/pull/784 is more important though I think )

Reply:
Filed https://github.com/hashgraph/hedera-sdk-js/pull/804